### PR TITLE
[None][test] validate GLM-5.2 feature support matrix

### PR DESCRIPTION
**Pull request by cascade812**
## Dev Engineer Review

- Added GLM-5 and GLM-5.2 accuracy coverage for TEP, DEP, MTP index sharing, acceptance rate, and runtime features. The expensive attention-DP, chunked-prefill, KV-cache-reuse, logits-processing, and guided-decoding checks share one model load.
- Updated supported-model documentation and the `GlmMoeDsaForCausalLM` feature matrix.
- Relocated GLM tests and updated related metadata and test paths.
- Updated CI manifests, QA lists, and waivers.
- Retained the TEP NVBug waiver and removed the unsupported LLGuidance test and its obsolete waiver.
- No unintended public API changes were reported.

## QA Engineer Review

- Added `TestGLM5FP8` and `TestGLM52NVFP4` test classes.
- Added coverage for GSM8K accuracy, TEP, attention-DP metrics, MTP index sharing, MTP acceptance rate, chunked prefill, KV-cache reuse, logits processing, and XGrammar guided decoding.
- Added `test_dep`, `test_8gpus`, `test_runtime`, and `test_mtp_index_share`; the MTP index-sharing test also validates aggregate draft-token acceptance.
- CI coverage is listed in `tests/integration/test_lists/test-db/l0_dgx_b200.yml`.
- Manual QA coverage is listed in `tests/integration/test_lists/qa/llm_function_core.txt`.
- Known skipped cases are listed in `tests/integration/test_lists/waives.txt`.
- Test-duration metadata uses the renamed `TestGLM52NVFP4` path.
- Verdict: needs follow-up because the L0 pipeline was unstable, the required multi-GPU approval label was missing, and CBTS coverage data was not provided.

## Description

- Add a dedicated GLM-5/GLM-5.2 accuracy suite covering TEP, DEP, MTP index sharing and acceptance rate, and the documented GLM feature-support matrix.
- Consolidate the expensive GLM-5.2 runtime checks for attention DP, KV-cache reuse, chunked prefill, logits processing, and guided decoding into `TestGLM52NVFP4::test_runtime`. These checks share one `LLM` instance so the GLM-5.2 checkpoint is loaded only once.
- Use the supported XGrammar backend for guided decoding. LLGuidance is not included because it is not supported for GLM-5.2 yet; the obsolete LLGuidance test entry and waiver were removed.
- Set the merged runtime test timeout to 15 minutes, allowing for cold checkpoint loading and CI variance without permitting a two-hour hang.
- Add GLM-5.2 to the supported-model list and document the `GlmMoeDsaForCausalLM` feature-support row.
- Move the existing GLM tests out of `test_llm_api_pytorch.py` and update CI lists, duration metadata, performance references, and waives.

## Test Coverage

Validated on 8x NVIDIA B200 using `tensorrt_llm==1.3.0rc26+ngcpytorch2605`:

Passed:
- `TestGLM52NVFP4::test_runtime`
  - all eight attention-DP ranks executed real context and generation work
  - the warm request reused KV-cache blocks while matching the cold request output
  - the 768-token prompt executed at least three context chunks
  - the custom logits processor forced the requested output token
  - XGrammar JSON-mode evaluation scored 100%
- `TestGLM52NVFP4::test_mtp_index_share`

Known limitation:
- LLGuidance guided decoding is not supported for GLM-5.2 and is intentionally excluded from the merged runtime test. https://nvbugspro.nvidia.com/bug/6700405
- `TestGLM52NVFP4::test_tep` is waived by NVBug 6668807.
- `TestGLM5FP8::test_8gpus` https://nvbugspro.nvidia.com/bug/6707362

Static validation:
- `python3 -m py_compile tests/integration/defs/accuracy/test_glm52.py`
- `python3 scripts/check_test_list.py --validate`
- `git diff --check`

## PR Checklist

- [x] PR description explains what and why.
- [x] Changes follow the coding guidelines.
- [x] Test coverage is included for the documented feature claims.
- [x] Documentation and CI test lists are updated.
- [x] No API changes or new dependencies are introduced.

### CODE REVIEW

File: tests/integration/defs/accuracy/test_glm52.py [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+        assert time_breakdown is not None
+        context_chunks = time_breakdown.get("ctx_chunk_metrics")
+        assert isinstance(context_chunks, list)
+        assert len(context_chunks) >= 3
+
+    @skip_pre_blackwell
+    @skip_ray
+    @pytest.mark.skip_less_mpi_world_size(8)
+    @pytest.mark.threadleak(enabled=False)
+    def test_kv_cache_reuse(self) -> None:
```
[the remark this answers is not in the document]
Comment (BowenFu) [reply tests/integration/defs/accuracy/test_glm52.py]: +1

**Review by arysef — APPROVED** [review]

File: tests/integration/test_lists/waives.txt [hunk 2]
```diff
@@ -14,6 +14,8 @@ accuracy/test_disaggregated_serving.py::TestLlama3_1_8BInstruct::test_tp_pp_symm
 accuracy/test_disaggregated_serving.py::TestQwen3_30B_A3B::test_mixed_ctx_gen_model[ctxpp2gentp2] SKIP (https://nvbugs/5748664)
 accuracy/test_dwdp_aggregated.py::TestDwdpAggDeepSeekV3Lite::test_dwdp_agg_accuracy[mode_a_uniform] SKIP (https://nvbugs/6661863)
 accuracy/test_dwdp_aggregated.py::TestDwdpAggDeepSeekV3Lite::test_dwdp_agg_accuracy[mode_a_uniform_contention_opt] SKIP (https://nvbugs/6644489)
+accuracy/test_glm52.py::TestGLM52NVFP4::test_guided_decoding[llguidance] SKIP (https://nvbugs/6700405)
+accuracy/test_glm52.py::TestGLM52NVFP4::test_tep[tp_size=8-ep_size=8] SKIP (https://nvbugs/6668807)
```
Comment (brnguyen2) [inline tests/integration/test_lists/waives.txt]: This waiver is carried over from `TestGLM52::test_nvfp4`, but the renamed `test_tep` no longer contains the `assert llm.args.kv_cache_config.use_kv_cache_manager_v2 is True` check the old test had (removed in this PR). If that assertion was the failure tracked by nvbugs/6668807, then (a) the waiver is likely stale — the renamed test may pass now and would stay skipped indefinitely, and (b) the config check the bug tracked no longer exists anywhere, so it can't regress-fail. Please either re-run `test_tep` and drop the waiver, or keep the check in a corrected form (e.g. assert the resolved value rather than the raw field) so the bug still has a detector.

File: docs/source/models/supported-models.md [hunk 3]
```diff
@@ -74,7 +74,7 @@ Note: Support for other models may vary. Features marked "N/A" are not applicabl
...
 | `DeepseekV32ForCausalLM`         | Yes               | Yes        | Yes                        | Yes                   | Yes             | Yes | No               | No                | No     | Yes           | Yes              | Yes            | N/A                      | Yes                   | Yes             |
-| `GlmMoeDsaForCausalLM`           | Yes               | Yes        | Untested                   | Yes                   | Yes             | Yes | No               | No                | No     | Yes           | Untested         | Untested       | N/A                      | Untested              | Untested        |
+| `GlmMoeDsaForCausalLM`           | Yes               | Yes        | Yes                        | Yes                   | Yes             | Yes | No               | No                | No     | Yes           | No               | Yes            | N/A                      | Yes                   | Yes             |
```
Comment (brnguyen2) [inline docs/source/models/supported-models.md:77]: Guided Decoding is flipped to `Yes` here, but this same PR waives `test_guided_decoding[llguidance]` as failing (nvbugs/6700405) — only the xgrammar backend is validated. Other rows in this table use footnotes for qualified support; consider `Yes [^n]` with a note that llguidance is currently not working for this model, otherwise the matrix overstates what a user can rely on.

File: tests/integration/defs/accuracy/test_glm52.py [hunk 4]
```diff
@@ -0,0 +1,550 @@
...
+                    max_tokens=output_length,
+                    temperature=0,
+                    end_id=-1,
+                ),
+                scheduling_params=scheduling_params,
+                use_tqdm=False,
+            )
+            stats_entries = llm.get_stats(timeout=10)
+            task = GSM8K(self.MODEL_NAME)
+            task.evaluate(llm)
```
Comment (brnguyen2) [inline tests/integration/defs/accuracy/test_glm52.py]: This full GSM8K eval runs under a config chosen for deterministic per-rank metric checks — no CUDA graphs, overlap scheduler disabled, autotuner off, `max_batch_size=8`, `free_gpu_memory_fraction=0.4` — which makes it far slower than the same eval in `test_tep`/`test_mtp_index_share` and is presumably why this test needs `timeout(7200)`. The ADP token/stats assertions don't depend on it, and accuracy under ADP isn't meaningfully validated by a config this constrained. Suggest dropping the eval from this test (or moving an ADP accuracy check into a separate test with a realistic batch/graph config).

File: tests/integration/defs/accuracy/test_glm52.py [hunk 5]
```diff
@@ -0,0 +1,550 @@
...
+            **pytorch_config,
+        ) as llm:
+            assert llm.args.quant_config.quant_algo == QuantAlgo.NVFP4
+            task = GSM8K(self.MODEL_NAME)
+            task.evaluate(llm)
+
+    @skip_pre_blackwell
+    @pytest.mark.skip_less_mpi_world_size(8)
+    @parametrize_with_ids("tp_size,ep_size", [(8, 8)])
+    def test_mtp_index_share_mtp_ar(self, tp_size, ep_size):
```
Comment (brnguyen2) [inline tests/integration/defs/accuracy/test_glm52.py]: This test boots an 8-GPU engine with a config identical to `test_mtp_index_share` (same MoE backend, MTP `max_draft_len=3`, chunked prefill, CUDA graph settings) just to measure acceptance rate on 3 prompts. Since both are in the QA list, that's a second full GLM-5.2 model load purely for a ~1-minute generation. Consider folding the acceptance-rate check into `test_mtp_index_share` after `task.evaluate(llm)` — same LLM instance, one boot instead of two.

**Review by brnguyen2 — COMMENTED** [review]
The move itself is complete — test-db, QA list, waives, durations, and the perf-config comment all track the new node IDs, and the perf-metrics-based acceptance-rate rewrite is more reliable than the old streaming token-count approach.

Two things worth resolving before merge, both detailed inline:

1. The nvbugs/6668807 waiver is carried to `test_tep`, but this same PR deletes the `use_kv_cache_manager_v2 is True` assertion that `test_nvfp4` had. If that assert was the failure behind the waiver, the renamed test may now pass while staying skipped indefinitely — and the config check itself is gone. Please re-validate and either drop the waiver or restore the check in corrected form.
2. The feature matrix flips Guided Decoding to "Yes" in the same PR that waives `test_guided_decoding[llguidance]` as failing (nvbugs/6700405). A footnote qualifying which backend is validated would keep the docs honest.

Minor: `test_attention_dp` runs a full GSM8K eval under a config tuned for deterministic metric checks (no CUDA graphs, no overlap scheduler, autotuner off, `max_batch_size=8`) — that's what forces the 7200s timeout, and the accuracy coverage duplicates `test_tep`/`test_mtp_index_share`. Consider dropping the eval from that test.

`[None]` is fine for a test-only change, and the description matches the diff.

**Review by StanleySun639 — APPROVED** [review]

**Review by chzblych — APPROVED** [review]

File: tests/integration/test_lists/waives.txt [hunk 6]
```diff
@@ -14,6 +14,7 @@ accuracy/test_disaggregated_serving.py::TestLlama3_1_8BInstruct::test_tp_pp_symm
 accuracy/test_disaggregated_serving.py::TestQwen3_30B_A3B::test_mixed_ctx_gen_model[ctxpp2gentp2] SKIP (https://nvbugs/5748664)
 accuracy/test_dwdp_aggregated.py::TestDwdpAggDeepSeekV3Lite::test_dwdp_agg_accuracy[mode_a_uniform] SKIP (https://nvbugs/6661863)
 accuracy/test_dwdp_aggregated.py::TestDwdpAggDeepSeekV3Lite::test_dwdp_agg_accuracy[mode_a_uniform_contention_opt] SKIP (https://nvbugs/6644489)
+accuracy/test_glm52.py::TestGLM52NVFP4::test_tep[tp_size=8-ep_size=8] SKIP (https://nvbugs/6668807)
```
Comment (BowenFu) [inline tests/integration/test_lists/waives.txt]: Could you link the current failing run for this exact GLM-5.2 TEP case (hardware, commit, and failure) in the PR description or NVBug? The new skip currently has only the bug ID, so its present scope cannot be verified. This evidence is required for this PR.

Comment (cascade812) [reply tests/integration/test_lists/waives.txt]: This test was migrated from accuracy/test_llm_api_pytorch.py::TestGLM52::test_nvfp4[tp_size=8-ep_size=8], was keeping it intentionally because the ticket is still open.  
As I just checked, it has been just resolved on main branch, will remove it and update the ticket status.

**Review by BowenFu — APPROVED** [review]

**Comment by schetlur-nv** [comment]
Should we move all this testing to GLM 5.3 instead, given that model has been released and is the same arch?

**Comment by cascade812** [comment]
> Should we move all this testing to GLM 5.3 instead, given that model has been released and is the same arch?

Good point, currently GLM5.3 nvfp4 is not available for CI yet, will update it to GLM 5.3 once it's available.

**Review by BowenFu — APPROVED** [review]
